commit b881881dbd38a1720a9f8401fd1baafdd0dfccc0
Author: wm4 <wm4@nowhere>
Date:   Tue Sep 19 18:33:37 2017 +0200

Revert "vd_lavc: change auto-probe order to prefer cuda over vdpau-copy"

This reverts commit 96462040ec79b353457b64949f96fad30bd6e988.

I guess the autoprobing is still too primitive to handle this well. What
it really should be trying is initializing the wrapper decoder, and if
that doesn't work, try another method. This is complicated by hwaccels
initializing in a delayed way, so there is no easy solution yet.

Probably fixes #4865.

diff --git a/video/decode/vd_lavc.c b/video/decode/vd_lavc.c
index 476beeb9e..e08246a29 100644
--- a/video/decode/vd_lavc.c
+++ b/video/decode/vd_lavc.c
@@ -261,11 +261,16 @@ static const struct vd_lavc_hwdec *const hwdec_list[] = {
 #endif
 #if HAVE_VDPAU_HWACCEL
     &mp_vd_lavc_vdpau,
+    &mp_vd_lavc_vdpau_copy,
 #endif
 #if HAVE_VIDEOTOOLBOX_HWACCEL
     &mp_vd_lavc_videotoolbox,
     &mp_vd_lavc_videotoolbox_copy,
 #endif
+#if HAVE_VAAPI_HWACCEL
+    &mp_vd_lavc_vaapi,
+    &mp_vd_lavc_vaapi_copy,
+#endif
 #if HAVE_D3D_HWACCEL
     &mp_vd_lavc_d3d11va,

@@ -285,13 +290,6 @@ static const struct vd_lavc_hwdec *const hwdec_list[] = {
     &mp_vd_lavc_cuda_old,
  #endif
     &mp_vd_lavc_cuda_copy,
-#endif
-#if HAVE_VDPAU_HWACCEL
-    &mp_vd_lavc_vdpau_copy,
-#endif
-#if HAVE_VAAPI_HWACCEL
-    &mp_vd_lavc_vaapi,
-    &mp_vd_lavc_vaapi_copy,
 #endif
     &mp_vd_lavc_crystalhd,
     NULL
